src/gpu/nvidia/cudnn_batch_normalization_executor.hpp

```diff
@@ -40,155 +41,159 @@ struct bnorm_exec_base_t {
     virtual ~bnorm_exec_base_t() = default;
 
 protected:
-    template <typename T, ::sycl::access::mode md, typename sc_t>
-    void *mean_var_ptr(::sycl::accessor<T, 1, md> acc, sc_t &sc,
-            const compat::interop_handle &ih) const {
-        return sc.template memory<void *>(ih, acc);
-    }
-
-    template <typename sc_t>
-    std::nullptr_t mean_var_ptr(std::nullptr_t acc, sc_t &,
-            const compat::interop_handle &ih) const {
-        return acc;
-    }
-
-    template <typename read_acc_t, typename write_acc_t, typename wkspace_st_t,
-            typename float_acc_t, typename maybe_nullptr_t>
+    template <::sycl::access::mode flt_m,
+            ::sycl::access::mode mean_var_m = ::sycl::access::mode::write>
     void interop_task_fwd(
             std::shared_ptr<cudnn_batch_normalization_impl_base_t> bnorm_impl,
             engine_t *engine, ::sycl::handler &cgh,
-            nvidia::sycl_cuda_stream_t *cuda_stream, read_acc_t src_acc,
-            write_acc_t dst_acc, maybe_nullptr_t mean_acc,
-            maybe_nullptr_t var_acc, float_acc_t scale_acc,
-            float_acc_t bias_acc, wkspace_st_t wkspace_st, bool init_ss,
-            bool init_mean_var) const {
-
-        std::shared_ptr<
-                ::sycl::accessor<uint8_t, 1, ::sycl::access::mode::write>>
-                wkspace_acc;
-        if (!wkspace_st->is_null()) {
-            wkspace_acc.reset(new ::sycl::accessor<uint8_t, 1,
-                    ::sycl::access::mode::write>(
-                    utils::downcast<sycl::sycl_buffer_memory_storage_t *>(
-                            wkspace_st)
-                            ->buffer()
-                            .template get_access<::sycl::access::mode::write>(
-                                    cgh)));
-        }
-
-        maybe_init_mean_var(cuda_stream, mean_acc, var_acc, init_mean_var);
-        maybe_init_ss(cuda_stream, scale_acc, bias_acc, init_ss);
+            nvidia::sycl_cuda_stream_t *cuda_stream,
+            sycl_memory_arg_t<::sycl::access::mode::read> arg_src,
+            sycl_memory_arg_t<::sycl::access::mode::write> arg_dst,
+            sycl_memory_arg_t<flt_m> arg_scale_bias,
+            sycl_memory_arg_t<::sycl::access::mode::write> arg_wkspace,
+            bool init_ss, bool init_mv,
+            sycl_memory_arg_t<mean_var_m> arg_mean
+            = sycl_memory_arg_t<::sycl::access::mode::write>(),
```

Comment:
Since access mode is a template parameter it can potentially change so I'd suggest to just use `= {}` for the default parameter.

Reply:
Change made